### PR TITLE
Transfer auth system to this repository

### LINKED ISSUE #1: Transferring auth system to this repository

### PR DESCRIPTION
**Pull request by FunctionalMetatable**
Resolves #1 

In this PR, I rewrote the authorization module. There is now a function `authMiddleware` that only lets users that are authenticated access the route.

This PR adds the routes:
GET `/users/:user` <= Returns a User
GET `/tutorials/:id` <= Returns a Tutorial that can be used to render the tutorial in the frontend
GET `/auth/me` <= Returns a User (the authenticated user) (Needs Authorization)
PUT `/tutorial/new` <= Creates a tutorial (Needs Authorization)

### CODE REVIEW

File: src/auth.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+const Database = require("@replit/database")
+const fetch = require("node-fetch")
+const crypto = require("crypto")
```
Comment (TheColaber) [inline src/auth.js:3]: This doesn't seem to be in the package.json

File: src/index.js [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
+const cookieParser = require('cookie-parser')
+const auth = require('./auth.js')
+const start = (new Date()).getTime()
+
+/* >>>>> SETTINGS <<<<< */
+
 const port = 3000;
 const frontendURL = "448me.sse.codesandbox.io";
-const backEndURL = "hly3v.sse.codesandbox.io";
+const backendURL = "Backend.9gr.repl.co";
```
Comment (TheColaber) [inline src/index.js:12]: u can changes this after we are ready to merge

File: package.json [hunk 3]
```diff
@@ -7,11 +7,13 @@
     "start": "nodemon src/index.js"
   },
   "dependencies": {
+    "@replit/database": "^2.0.1",
```
Comment (TheColaber) [inline package.json:10]: TODO: Move to mongodb so this works on other hosters

**Review by TheColaber — CHANGES_REQUESTED** [review]
epic code man

File: src/auth.js (hunk 1 above)
Comment (FunctionalMetatable) [reply src/auth.js:3]: crypto is a built in module

File: src/index.js (hunk 2 above)
Comment (FunctionalMetatable) [reply src/index.js:12]: yes

File: package.json (hunk 3 above)
Comment (FunctionalMetatable) [reply package.json:10]: can't we keep the same module and switch databases according to where we are hosting on?

Comment (TheColaber) [reply package.json:10]: does localhost have a database? I'm not sure exactly what you are thinking

File: src/auth.js (hunk 1 above)
Comment (TheColaber) [reply src/auth.js:3]: As long as thats the case everywhere, I'm fine then

File: package.json (hunk 3 above)
Comment (FunctionalMetatable) [reply package.json:10]: I meant using different databases for different types of hosting on a use case, but it seems impractical. (I'll try start on mongodb)

Comment (TheColaber) [reply package.json:10]: You can do it in a separate PR if you want so I can merge this

Comment (TheColaber) [reply package.json:10]: Alright, merging

**Pull request merged**